### ISSUE TITLE
Report pageData["firstPaint"] for Firefox's timeToNonBlankPaint and IE/Edge's msFirstPaint

### ISSUE THREAD

**Issue by cpeterso** [WPO-Foundation/wptagent#30]
Firefox Nightly 56 just added a `window.performance.timing.timeToNonBlankPaint` ([Firefox bug 1377251](https://bugzilla.mozilla.org/show_bug.cgi?id=1377251))
 that is akin to Chrome's `window.chrome.loadTimes().firstPaintTime`. And IE and Edge have `window.performance.timing.msFirstPaint` ([MSDN](https://msdn.microsoft.com/en-us/library/ff974719(v=vs.85).aspx)).

wptagent can use Firefox's `timeToNonBlankPaint` and IE/Edge's `msFirstPaint` to report `pageData["firstPaint"]` like it does for chromeTimes["firstPaintTime"] (in [page_data.js#L25-L32](https://github.com/WPO-Foundation/wptagent/blob/5fd725dcb2d4d1251441e516fc2f618c5b644b1e/internal/js/page_data.js#L25-L32)).

Note that Firefox's `timeToNonBlankPaint` is not standard and requires that the Firefox user profile set the about:config pref `dom.performance.time_to_non_blank_paint.enabled` = true. Also `timeToNonBlankPaint`, unlike the other `performance.timing` properties, is the millisecond duration since `performance.timing.navigationStart`, not a timestamp relative to the UNIX epoch.

CC @digitarald

**Comment by pmeenan** [comment]
Just [landed the code](https://github.com/WPO-Foundation/wptagent/commit/acc4e84e11adb9aca5e28da5be919be715ae7437) and [appears to be working](https://www.webpagetest.org/result/170721_Y9_40ed4317d2b8efd54a27e783e1eb0284/1/details/#waterfall_view_step1).

While I was in there I fixed the fact that it wasn't extracting any of the user timing, performance.timing or custom metrics so now you should be able to collect arbitrary metrics as well.

This doesn't handle the case of the legacy (wptdriver) agent though.  Should be easy to land there as well, just requires a bit more work since the extension there needs to be built and signed.

**Comment by cpeterso** [comment]
@pmeenan Is it expected that the test details page's Waterfall View header lists "msFirstPaint" (a timer only available in IE/Edge) for Firefox and Chrome results?

**Comment by pmeenan** [comment]
Nope, fixed.  Apparently the data table had been updated a long time ago (to "RUM First Paint") but the legend for the waterfall hadn't.  They should match now.

**Comment by cpeterso** [comment]
@pmeenan, please note that the meaning of Firefox's performance.timing.timeToNonBlankPaint property has changed from "duration since navigationStart" to "timestamp relative to the UNIX epoch" (like all the other performance.timing.* properties).

https://hg.mozilla.org/integration/mozilla-inbound/rev/7a7db98b3785

**Comment by pmeenan** [comment]
Looks like it hasn't quite rolled out to firefox-trunk builds yet.  I have the change staged and ready to go but I'll wait until I can test it before pushing.
